### PR TITLE
Add middleware from Satellite

### LINKED ISSUE #41: Add middleware from Satellite
## Description
Add the following middleware to the `app`:
- Logger (using Pino)
- cors
- helmet
- errorHandler

**Resources**
- https://github.com/pinojs/pino-http
- https://www.npmjs.com/package/cors
- https://helmetjs.github.io
- https://github.com/Seneca-CDOT/telescope/tree/master/src/satellite

### PR DESCRIPTION
**Pull request by SerpentBytes**
## Description

This PR is a continuation of #89 and resolves #41 by adding the following middleware:

- [x]  helmet
- [x]  cors
- [x]  pino

### Steps to Test:
- Run the project locally
- Look for the following `log.info` message: 
```
{"level":30,"time":1675382308133,"pid":12627,"hostname":"TD","msg":"✅ app ready: http://localhost:8080"}
```
### Additional Information

I have deleted "useful headers" being set by Remix Blues Stack, as `helmet` by default adds that header. 
We can use `logger` from `app/lib/logger.server.ts` to `log` info, debug, warn, trace, and error level messages.

### CODE REVIEW

**Comment by humphd** [comment]
Let me know when this is ready for review.

File: app/satellite/logger.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import pino from 'pino';
```
Comment (humphd) [inline app/satellite/logger.ts:1]: Let's move this file to `app/lib/logger.server.ts`

File: app/satellite/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import type express from 'express';
```
Comment (humphd) [inline app/satellite/middleware.ts:1]: I think we should consider removing this, and using Remix's error handling.  It has a whole bunch of stuff we can leverage.

File: server.ts [hunk 3]
```diff
@@ -24,23 +32,24 @@ app.use((req, res, next) => {
 app.use(compression());
 
 // http://expressjs.com/en/advanced/best-practice-security.html#at-a-minimum-disable-x-powered-by-header
-app.disable('x-powered-by');
+app.disable("x-powered-by");
```
Comment (humphd) [inline server.ts:35]: Remove this, Helmet does this for us.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Review by dadolhay — APPROVED** [review]

**Pull request merged**